src/gtk/srchctrl.cpp

```diff
@@ -0,0 +1,429 @@
+
+    if ( !wx_is_at_least_gtk3(6) )
+    {
+        gtk_entry_set_icon_from_icon_name(m_entry, 
+                                          GTK_ENTRY_ICON_PRIMARY,
+                                          "edit-find-symbolic");
+
+        // Mimic the behaviour of the native GtkSearchEntry.
+        gtk_entry_set_icon_sensitive(m_entry, GTK_ENTRY_ICON_PRIMARY, FALSE);
+        gtk_entry_set_icon_activatable(m_entry, GTK_ENTRY_ICON_PRIMARY, FALSE);
+    }
+
+    g_signal_connect(m_entry, "icon-press", G_CALLBACK(wx_gtk_icon_press), this);
+}
+
+GtkEditable *wxSearchCtrl::GetEditable() const
+{
+    return GTK_EDITABLE(m_entry);
+}
+
+void wxSearchCtrl::Clear()
+{
+    wxTextEntry::Clear();
+    ShowCancelButton(false);
+}
+
+bool wxSearchCtrl::IsModified() const
+{
+    return m_modified;
+}
+
+void wxSearchCtrl::MarkDirty()
+{
+    m_modified = true;
+}
+
+void wxSearchCtrl::DiscardEdits()
+{
+    m_modified = false;
+}
+
+bool wxSearchCtrl::PositionToXY(long pos, long *x, long *y ) const
+{
+    if (pos <= GTKGetEntryTextLength(GetEntry()))
+    {
+        if ( y )
+            *y = 0;
+        if ( x )
+            *x = pos;
+    }
+    else
+    {
+        // index out of bounds
+        return false;
+    }
+
+    return true;
+}
+
+long wxSearchCtrl::XYToPosition(long x, long y ) const
+{
+    if ( y != 0 || x > GTKGetEntryTextLength(GetEntry()) )
+        return -1;
+
+    return x;
+}
+
+int wxSearchCtrl::GetLineLength(long lineNo) const
+{
+    const wxString str = GetLineText(lineNo);
+    return (int) str.length();
+}
+
+int wxSearchCtrl::GetNumberOfLines() const 
+{
+    return 1;
+}
+
+wxString wxSearchCtrl::GetLineText( long lineNo ) const
+{
+    if ( lineNo == 0 )
+        return GetValue();
+
+    return wxString();
+}
+
+void wxSearchCtrl::ShowPosition( long pos )
+{
+    gtk_editable_set_position(GetEditable(), pos);
+}
+
+// search control specific interfaces
+// ----------------------------------
+#if wxUSE_MENUS
+
+void wxSearchCtrl::SetMenu( wxMenu* menu )
+{
+    if ( menu == m_menu )
+    {
+        // no change
+        return;
+    }
+
+    delete m_menu;
+    m_menu = menu;
+
+    const bool hasMenu = m_menu != NULL;
+
+    gtk_entry_set_icon_sensitive(m_entry, GTK_ENTRY_ICON_PRIMARY, hasMenu);
+    gtk_entry_set_icon_activatable(m_entry, GTK_ENTRY_ICON_PRIMARY, hasMenu);
+}
+
+wxMenu* wxSearchCtrl::GetMenu()
+{
+    return m_menu;
+}
+
+#endif // wxUSE_MENUS
+
+void wxSearchCtrl::ShowSearchButton(bool WXUNUSED(show))
+{
+}
+
+bool wxSearchCtrl::IsSearchButtonVisible() const
+{
+    return HasMenu();
+}
+
+void wxSearchCtrl::ShowCancelButton(bool show)
+{
+#if wxHAS_GTK_SEARCH_ENTRY
+    // The cancel button is shown/hidden automatically by the GtkSearchEntry.
+    if ( wx_is_at_least_gtk3(6) )
+        return;
+#endif // wxHAS_GTK_SEARCH_ENTRY
+
+    if ( show == IsCancelButtonVisible() )
+    {
+        // no change
+        return;
+    }
+
+    if ( show && !m_cancelButtonVisible )
```

Comment:
```suggestion
    if ( show )
```
Because the other part of the test looks redundant, as we already checked for the "no change" case above.

Also, I'd probably just write this as
```cpp
gtk_entry_set_icon_from_icon_name(m_entry,
                                  GTK_ENTRY_ICON_SECONDARY,
                                  show ? "edit-clear-symbolic" : NULL); 
```